tmt/cli.py

```diff
@@ -2081,3 +2081,100 @@ def completion_fish(context: Context, install: bool, **kwargs: Any) -> None:
     Setup shell completions for fish.
     """
     setup_completion('fish', install)
+
+
+@click.command(context_settings={
+    "ignore_unknown_options": True,
+    "allow_extra_args": True,
+    })
+@click.pass_context
+@click.option(
+    '-r', '--root', metavar='PATH', show_default=True, default='.',
+    help='Variable passed to tmt main. See `tmt --help`')
+@click.option(
+    '-v', '--verbose', count=True, default=0,
+    help='Variable passed to tmt main. See `tmt --help`')
+@click.option(
+    '-d', '--debug', count=True, default=0,
+    help='Variable passed to tmt main. See `tmt --help`')
+@click.option(
+    '--no-color', is_flag=True, default=False,
+    help='Variable passed to tmt main. See `tmt --help`'
+    )
+@click.option(
+    '--force-color', is_flag=True, default=False,
+    help='Variable passed to tmt main. See `tmt --help`'
+    )
+@click.option(
+    '--version', is_flag=True,
+    help="Variable passed to tmt main. See `tmt --help`")
+@click.option(
+    '--lint-what', default=None,
+    type=click.Choice(['tests', 'plans', 'stories']),
+    help='tmt types to lint')
+def pre_commit(
+        context: click.Context,
+        root: str,
+        verbose: int,
+        debug: int,
+        version: bool,
+        no_color: bool,
+        force_color: bool,
+        lint_type: Optional[str]) -> None:
+    """
+    Cli wrapper of tmt lint.
+    """
+    # Special handling for --version
+    if version:
+        main(['--version'])
+
+    # Create logger to reuse tmt.utils.Command utility
+    # apply_colors_output is not used because this is recreated and handled by main app instead
+    apply_colors_output, apply_colors_logging = tmt.log.decide_colorization(no_color, force_color)
+    logger = tmt.log.Logger.create(
+        apply_colors_output=apply_colors_output,
+        apply_colors_logging=apply_colors_logging
+        )
+    logger.add_console_handler()
```

Comment:
I have unresolved [this thread](https://github.com/teemtee/tmt/pull/2007#discussion_r1186445840) because there is some context on why we had to add the logger here and why the workflow was changed from simple non-click entry-point to a `click.command`.